### DISCUSSION TITLE
explanations/details  of your  algorithms, for example  how Multi-collinearity is removed?

### DISCUSSION THREAD

**Discussion by Sandy4321** [pycaret/pycaret#1283] [Q&A]
great package thanks!!!
1
do you have explanations/details of your algorithms?
2
for example how Multi-collinearity is removed?
2.2
Multi-collinearity: Multi-collinearity is a state of very high intercorrelations or inter-associations among the independent features in the dataset. It is a type of disturbance in the data that is not handled well by machine learning models (mostly linear algorithms). Multi-collinearity may reduce overall coefficient of the model and cause unpredictable variance. This will lead to overfitting where the model may do great on a known training set but will fail with an unknown test set 

https://github.com/pycaret/pycaret/blob/master/Tutorials/Binary%20Classification%20Tutorial%20Level%20Intermediate%20-%20CLF102.ipynb

I see in 
https://github.com/pycaret/pycaret/blob/master/preprocess.py
## so in that case , we can simply pick one of the variable

but lets say
th = 0.8
corr(a,b) = 0.9
corr(a.c) = 0.92
corr(b,c) = 0.99

corr(a,d) = 0.6
corr(b,d) = 0.6
corr(c,d) = 0.87

then will be removed:
d  since  corr(c,d) = 0.87 > th 
and 
c  since corr(b,c) = 0.99 > th
and 
b since  corr(a,b) > th
so only feature "a"  will left?

2.3
also in which order you do remove features?

**Comment by Sandy4321** [comment]
any answer pls?
